web/src/message_view.js

```diff
@@ -662,7 +670,39 @@ export function show(raw_terms, opts) {
                 }
                 message_fetch.load_messages_for_narrow({
                     anchor,
-                    cont() {
+                    cont(data, opts) {
+                        const msg_in_narrow = data?.messages[0]?.id;
+                        let adjusted_terms;
+
+                        if (msg_in_narrow !== undefined) {
+                            adjusted_terms = adjusted_terms_if_moved(
+                                filter._terms,
+                                message_store.get(msg_in_narrow),
+                            );
+                        }
+
+                        if (adjusted_terms) {
+                            filter = new Filter([...adjusted_terms]);
+                            id_info = {then_select_id: data.anchor};
+                            opts.change_hash = true;
+                            const updated_msg_list_data = create_and_update_message_list(
+                                filter,
+                                id_info,
+                                opts,
+                            );
+
+                            msg_list = updated_msg_list_data.msg_list;
+                            restore_rendered_list = updated_msg_list_data.restore_rendered_list;
+                            select_immediately = restore_rendered_list;
+                            handle_post_message_list_change(
+                                id_info,
+                                msg_list,
+                                opts,
+                                select_immediately,
+                                select_opts,
+                                then_select_offset,
+                            );
+                        }
```

Comment:
Hmm, can you explain to me why and what are we doing here?

Reply:
No longer releavant since I removed this block of code.

Reply:
As per the CZO:
>-  If the message exists and is visible to the user, then replace any "conversation" filters — channel/stream, topic, DM conversation — with filters corresponding to the conversation the message is in.
>- Otherwise, ignore the with (go to the server directly without the recursive fetching of target message).

The above change occurs when the target_message is not available locally. So with `data` fetched from `load_messages_for_narrow`, (containing messages fetched for the `with` operator, and anchor position), I try to create a message list (and change hash) if the terms are adjusted (i.e., the current narrow, and the first message returned from server are of same narrow) and then call  the `handle_post_view_change`.

If the terms message fetched from server, and the current narrow are of same narrow, then none of these is executed.

Reply:
Yeah, we do this now in message_fetch now.

```
+            if (opts.validate_filter_topic_post_fetch) {
+                opts.msg_list.data.filter.try_adjusting_for_moved_with_target(messages[0]);
+            }

```